src/main/scala/diplomacy/SRAM.scala

```diff
@@ -34,16 +32,43 @@ abstract class DiplomaticSRAM(
   def mask: List[Boolean] = bigBits(address.mask >> log2Ceil(beatBytes))
 
   // Use single-ported memory with byte-write enable
-  def makeSinglePortedByteWriteSeqMem(size: BigInt, lanes: Int = beatBytes, bits: Int = 8) = {
+  def makeSinglePortedByteWriteSeqMem(
+    size: BigInt,
+    lanes: Int = beatBytes,
+    bits: Int = 8,
+    busProtocol: Option[OMProtocol],
+    dataECC: Option[OMECC] = None,
+    hasAtomics: Option[Boolean] = None) = {
     // We require the address range to include an entire beat (for the write mask)
     val mem =  DescribedSRAM(
       name = devName.getOrElse("mem"),
       desc = devName.getOrElse("mem"),
       size = size,
-      data = Vec(lanes, UInt(width = bits))
+      data = Vec(lanes, UInt(width = bits)),
+      id = DescribedSRAMIdAssigner.genId()
     )
     devName.foreach(n => mem.suggestName(n.split("-").last))
 
+    val omSRAM: OMSRAM = DiplomaticObjectModelAddressing.makeOMSRAM(
+      desc = "mem", //lim._2.name.map(n => n).getOrElse(lim._1.name),
+      depth = size,
+      data = Vec(lanes, UInt(width = bits)),
+      id = DescribedSRAMIdAssigner.genId()
+    )
+
+    parentLogicalTreeNode.map {
+      case parentLTN =>
+        def sramLogicalTreeNode = new BusMemoryLogicalTreeNode(
+          device = device,
+          omSRAMs = Seq(omSRAM),
+          busProtocol = busProtocol.getOrElse(throw new IllegalArgumentException("Protocol not specified")),
+          dataECC = dataECC,
+          hasAtomics = hasAtomics,
+          busProtocolSpecification = None)
+        LogicalModuleTree.add(parentLTN, sramLogicalTreeNode)
+    }
+
+
```

Comment:
Why is there both OMMem and OMSRAM?

Reply:
Seems to me like you should unify them, return it/them, and add them to the logical tree in the parent

Reply:
OMMem is legacy and can be removed soon.

Reply:
OMMem is used and will be replaced in another PR